autoemulate/experimental/emulators/base.py

```diff
@@ -168,3 +168,29 @@ def get_tune_config():
             "each subclass."
         )
         raise NotImplementedError(msg)
+
+
+class SklearnBackend(Emulator):
+    """
+    SklearnBackend is a sklearn model and implements the base class.
+    This provides default implementations to further subclasses.
+    This means that models can subclass and only need to implement
+    `.fit()` and `.predict()` to have an emulator to be run in `AutoEmulate`
+    """
+
```

Comment:
Could add an attribute here with a type such as: `class SklearnEstimator(BaseEstimator, RegressorMixin): ...`. This could be referenced in default impls of the backend.

```suggestion
    model_: SklearnEstimator
```

Reply:
```
class SklearnEstimator(BaseEstimator, RegressorMixin): ...

class SklearnBackend(Emulator):
    """
    SklearnBackend is a sklearn model and implements the base class.
    This provides default implementations to further subclasses.
    This means that models can subclass and only need to implement
    `.fit()` and `.predict()` to have an emulator to be run in `AutoEmulate`
    """

    model: SklearnEstimator

    def __init__(
        self, x: InputLike | None = None, y: InputLike | None = None, **kwargs
    ):
        pass
```